Snapshot publishing to sonatype fails

https://github.com/kotest/kotest/actions/runs/9915068638/job/27396015667

```
Execution failed for task ':kotest-bom:publishKotestBomPublicationToDeployRepository'.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> Failed to publish publication 'KotestBom' to repository 'deploy'
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/kotest-bom/5.10.0.1526-SNAPSHOT/maven-metadata.xml'. Received status code 401 from server: Content access is protected by token
```

I think Sonatype broke publishing https://community.sonatype.com/t/401-content-access-is-protected-by-token-authentication-failure-while-performing-maven-release/12741

_Originally posted by @aSemy in https://github.com/kotest/kotest/issues/4012#issuecomment-2233323505_

The Sonatype credentials are fixed, but now it looks like the Gradle config is broken. 

https://github.com/kotest/kotest/actions/runs/10019559908/job/27696885243#step:7:167

```
> Task :kotest-assertions:kotest-assertions-shared:compileKotlinAndroidNativeArm32
FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
52 actionable tasks: 44 executed, 8 from cache
A problem was found with the configuration of task ':kotest-common:signAndroidNativeArm64Publication' (type 'Sign').
  - Gradle detected a problem with the following location: '/Users/runner/work/kotest/kotest/kotest-common/build/libs/kotest-common-5.10.0.1536-SNAPSHOT-javadoc.jar.asc'.
    
    Reason: Task ':kotest-common:publishAndroidNativeArm32PublicationToDeployRepository' uses this output of task ':kotest-common:signAndroidNativeArm64Publication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':kotest-common:signAndroidNativeArm64Publication' as an input of ':kotest-common:publishAndroidNativeArm32PublicationToDeployRepository'.
      2. Declare an explicit dependency on ':kotest-common:signAndroidNativeArm64Publication' from ':kotest-common:publishAndroidNativeArm32PublicationToDeployRepository' using Task#dependsOn.
      3. Declare an explicit dependency on ':kotest-common:signAndroidNativeArm64Publication' from ':kotest-common:publishAndroidNativeArm32PublicationToDeployRepository' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.7/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./
BUILD FAILED in 1m 48s
Publishing build scan...
https://gradle.com/s/cdlnlglnpnm4k
```

I'll re-open and re-use this issue, and look into a fix.

https://github.com/gradle/gradle/issues/26091

--- a/buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts
@@ -103,3 +103,11 @@ tasks.withType<PublishToMavenRepository>().configureEach {
    usesService(mavenPublishLimiter)
 }
 //endregion
+
+//region Fix Gradle error Reason: Task <publish> uses this output of task <sign> without declaring an explicit or implicit dependency.
+// https://github.com/gradle/gradle/issues/26091
+tasks.withType<AbstractPublishToMaven>().configureEach {
+   val signingTasks = tasks.withType<Sign>()
+   mustRunAfter(signingTasks)
+}
+//endregion

